File: 백준/Silver/1251. 단어 나누기/단어 나누기.py
import sys

input = sys.stdin.readline

'''
단어 나누기 (https://www.acmicpc.net/problem/1251)
'''

def solution(word):
    length = len(word)
    words = []
    for i in range(length):
        for j in range(i, length):
            prefix = word[:i]
            middle = word[i:j]
            postfix = word[j:]
            if prefix != '' and middle != '' and postfix != '':
                words.append(prefix[::-1] + middle[::-1] + postfix[::-1])

    return sorted(words)[0]

word = input().rstrip()
print(solution(word))

File: 백준/Silver/1251. 단어 나누기/단어 나누기.java
import java.io.*;
import java.util.*;

/**
 * https://www.acmicpc.net/problem/1251
 */
public class Main {

    public static String solution(String word) {
        int length = word.length();
        ArrayList<String> lst = new ArrayList<>();
        for (int i = 1; i < length - 2; i++) {
            for (int j = i + 1; j < length; j++) {
                StringBuilder sb = new StringBuilder();
                sb.append(new StringBuilder(word.substring(0, i)).reverse());
                sb.append(new StringBuilder(word.substring(i, j)).reverse());
                sb.append(new StringBuilder(word.substring(j)).reverse());
                lst.add(sb.toString());
            }
        }

        Collections.sort(lst);

        return lst.get(0);
    }

    public static void main(String[] args) throws IOException {
        InputStream input = System.in;
        BufferedReader br = new BufferedReader(new InputStreamReader(input));

        String word = br.readLine();
        System.out.println(solution(word));
    }
}
